### PR TITLE
A few small DUs to structs

### PR DESCRIPTION
**Pull request by Thorium**
Tiny optimisation only.

### CODE REVIEW

**Comment by Lanayx** [comment]
Hi, thanks for the PR,but have you checked the performance of struct DU changes? In my experience DU s with more than two cases usually perform better as classes than structs because of inefficient memory layout.

File: src/Oxpecker.ViewEngine/IdeTweaks.fs [hunk 1]
```diff
@@ -2,6 +2,7 @@ namespace JetBrains.Annotations
 
 open System
 
+[<Struct>]
```
Comment (Lanayx) [inline src/Oxpecker.ViewEngine/IdeTweaks.fs:5]: This is already a struct (because enum)

**Comment by Thorium** [comment]
The existing performance test, before this commit: 
image
after this commit:
image

I think the change is minimal or not hit by the existing performance tests?


Anyway, if you refer to https://www.bartoszsypytkowski.com/writing-high-performance-f-code/#discriminatedunionsasvaluetypes the DUs I changed don't carry any values. Rather e.g. HttpVerbs were used in a seq of verbs in part of another recursive DU, so I thought there is object-capsulation enough already, this could save some memory.

File: src/Oxpecker.ViewEngine/IdeTweaks.fs (hunk 1 above)
Comment (Thorium) [reply src/Oxpecker.ViewEngine/IdeTweaks.fs:5]: oh, yes it is

**Comment by Lanayx** [comment]
Thank you for the explanation, you are right. As for HttpVerbs - it's unclear whether it's more benefitial to leave it class of make struct because there is no perf test for building route map, current tests test already launched server, do you want to add that to the PR? It should easily justify the changes and provide the ground for future improvements.

**Comment by Thorium** [comment]
I tested to add

```fsharp
    [<Benchmark>]
    member this.OxpeckerSetupServer() = OxpeckerRouting.webApp().CreateClient()

    [<Benchmark>]
    member this.GiraffeSetupServer() = GiraffeRouting.webApp().CreateClient()
```
Current main:
image

This branch:
image

...but the differences are still within the error-margin. I wouldn't add this start-server perf-test to the repo because it undermines the purpose of this webserver: The most important figure should be the delivery speed. If your webserver start within a second, that's good enough. Unless you main focus is some serverless scenario.

**Review by Lanayx — APPROVED** [review]

**Comment by Lanayx** [comment]
Got it, thanks for checking!

**Pull request merged**